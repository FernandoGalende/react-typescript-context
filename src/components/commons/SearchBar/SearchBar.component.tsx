import React, {Fragment} from 'react';
import './SearchBar.scss';

type SearchBarProps = {
	onChange: any;
};

const SearchBar: React.FC<SearchBarProps> = ({onChange}) => {
	const searchInput = (evt: any) => {
		onChange(evt.target.value.toLowerCase());
	};

	return (
		<Fragment>
			<div className='search-container'>
				<div className='text-input'>
					<input type='text' id='input1' onChange={searchInput} />
					<label htmlFor='input1'>Search:</label>
				</div>
			</div>
		</Fragment>
	);
};

export default SearchBar;
