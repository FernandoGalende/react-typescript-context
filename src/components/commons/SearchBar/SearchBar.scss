.search-container {
	box-sizing: border-box;
	padding: 30px;
	display: flex;
	align-items: center;
	height: 10%;
}
