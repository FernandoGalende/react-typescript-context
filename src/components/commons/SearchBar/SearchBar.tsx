import React, {Fragment} from 'react';
import './SearchBar.scss';

type SearchBarProps = {
	onChange: Function;
};

export const SearchBar: React.FC<SearchBarProps> = ({onChange}) => {
	const searchInput = (event: React.ChangeEvent<HTMLInputElement>) => {
		onChange(event.target.value.toLowerCase());
	};

	return (
		<Fragment>
			<div className='search-container'>
				<div className='text-input'>
					<input data-testid='input-test' type='text' id='input' onChange={searchInput} />
					<label data-testid='label-test' htmlFor='input'>
						Search:
					</label>
				</div>
			</div>
		</Fragment>
	);
};
