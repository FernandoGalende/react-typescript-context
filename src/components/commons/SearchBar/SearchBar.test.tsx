import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {SearchBar} from './SearchBar';

const mockFuntion = jest.fn();

describe('<SearchBar />', () => {
	let getByTestId: Function;
	let container: HTMLElement;

	beforeEach(() => {
		const queries = render(<SearchBar onChange={mockFuntion}>click!</SearchBar>);
		container = queries.container;
		getByTestId = queries.getByTestId;
	});

	it('match snapshot', () => {
		expect(container).toMatchSnapshot();
	});

	it('should input exist', () => {
		expect(getByTestId('input-test')).toBeInTheDocument();
	});

	it('should label exist', () => {
		expect(getByTestId('label-test')).toHaveTextContent('Search:');
	});

	it('text on input should output function', async () => {
		const input = getByTestId('input-test');
		fireEvent.change(input, {target: {value: 'a'}});
		expect(mockFuntion).toHaveBeenCalledTimes(1);
	});

	it('text on input should have properly test', async () => {
		const input = getByTestId('input-test');
		fireEvent.change(input, {target: {value: 'a'}});
		expect(input.value).toEqual('a');
	});
});
