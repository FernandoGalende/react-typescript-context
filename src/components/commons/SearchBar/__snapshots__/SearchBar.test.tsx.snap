// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> match snapshot 1`] = `
<div>
  <div
    class="search-container"
  >
    <div
      class="text-input"
    >
      <input
        data-testid="input-test"
        id="input"
        type="text"
      />
      <label
        data-testid="label-test"
        for="input"
      >
        Search:
      </label>
    </div>
  </div>
</div>
`;
